test/e2e/nodepool_test.go

```diff
@@ -44,14 +49,45 @@ func TestNodePool(t *testing.T) {
 	// We set replicas to 0 in order to allow the inner tests to
 	// create their own NodePools with the proper replicas
 	clusterOpts.NodePoolReplicas = 0
-	guestCluster := e2eutil.CreateCluster(t, ctx, mgmtClient, &clusterOpts, globalOpts.Platform, globalOpts.ArtifactDir)
-	guestClient := e2eutil.WaitForGuestClient(t, ctx, mgmtClient, guestCluster)
+	hostedCluster := e2eutil.CreateCluster(t, ctx, mgmtClient, &clusterOpts, globalOpts.Platform, globalOpts.ArtifactDir)
+	guestClient := e2eutil.WaitForGuestClient(t, ctx, mgmtClient, hostedCluster)
+
+	// Get the newly created defautlt NodePool
+	nodepools := &hyperv1.NodePoolList{}
+	if err := mgmtClient.List(ctx, nodepools, crclient.InNamespace(hostedCluster.Namespace)); err != nil {
+		t.Fatalf("failed to list nodepools in namespace %s: %v", hostedCluster.Namespace, err)
+	}
+	if len(nodepools.Items) != 1 {
```

Comment:
what if the test implies HA? maybe we just should check that is not 0?

Reply:
AFAIK, the cli will only create 1 nodePool by default, no?

Reply:
yep, we need to have this in mind if the HostedCluster definition changes in the future.